### PR TITLE
fix(Scripts/Battlegrounds/AlteracValley): Adding missing Tower Archers

### PR DESCRIPTION
**Pull request by Knindzagxg**
## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).

## SOURCE:

The changes have been validated through:
- [x] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue

1. Log into game and .debug bg.
2. Check spawns on the Towers and Archers AI.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by icedcolor** [comment]
There are no NPC guards in Horde's cave.

File: src/server/game/Battlegrounds/Zones/BattlegroundAV.h
Comment (sudlud) [inline src/server/game/Battlegrounds/Zones/BattlegroundAV.h:1]: changing all the comments on these lines really makes it hard to actually see what you intended to change imo.

**Comment by sudlud** [comment]
> There are no NPC guards in Horde's cave.

Is this an issue of this PR or just a general issue of AV?

**Comment by Knindzagxg** [comment]
> There are no NPC guards in Horde's cave.

Will see if i can sort it in next updates.
Those are for missing Archers on the towers.

**Comment by Grimdhex** [comment]
I'm not sure about this PR. There are missing or just spawn but move elsewhere ?

**Comment by Knindzagxg** [comment]
> I'm not sure about this PR. There are missing or just spawn but move elsewhere ? And for the title you can just use : fix(Battlegrounds/AlteracValley)

There are no Archers on towers, like its supposed to be.
Instead, there are some random guards that just roam up there.

This one fix it, add archers like its supposed to be.

**Comment by sudlud** [comment]
Thanks for the feedback, so apparently there was a mismatch of the two enums CreatureInfo and CreatudeIds which is corrected in this PR, and also a no-move flag beeing set for the archers.

So before PR, the bowman were present in AV at grave defense spawn positions, whereas some normal guards were roaming at tower defense spawn positions.

So after PR, grave defense and tower archers should have the correct creature IDs assigned now.

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**